osu.Game/Rulesets/Mods/ModSuddenDeath.cs

```diff
@@ -23,7 +23,7 @@ public abstract class ModSuddenDeath : ModFailCondition
         public override Type[] IncompatibleMods => base.IncompatibleMods.Append(typeof(ModPerfect)).ToArray();
 
         protected override bool FailCondition(HealthProcessor healthProcessor, JudgementResult result)
-            => result.Type.AffectsCombo()
+            => (result.Judgement.MinResult.AffectsCombo() || result.Judgement.MaxResult.AffectsCombo() || result.Type.AffectsCombo())
```

Comment:
For this, I'm not completely sure if either the old _or_ the new code is correct, since a `ComboBreak` will cause a fail with Sudden Death active, even though it is technically not a "miss"...

Also, doing this check this way means that in osu! dropping a sliderend will now fail with Sudden Death. The previous code would only check that `MinResult` does not affect combo and bail. The new code will check that `MaxResult` affects combo and cause a failure. [If I am to trust wiki this does not happen on stable.](https://osu.ppy.sh/wiki/en/Gameplay/Game_modifier/Sudden_Death)

Reply:
Hmm okay. So:

osu:
- Sudden death doesn't fail on missed tails.
- Sudden death fails on missed tick.
- Perfect fails on both.

o!m:
- Sudden death fails on combo break.

So the original implementation sounds correct. There's no tests for this, maybe I'll add some...